Bad_Practices/Coding_Practies/CPBP.md

```diff
@@ -0,0 +1,21 @@
+## Mistakes to avoid when coding
+
+### 1.Poor Documentaion
+
+Developers tend to get caught up in the process of coding majority of the time that theu foten forget to document it. Often you will forget what your code does or how it does things or even the approach of hiw you came up with this solution. Including documentaion helps with reusability and maintenance of your code and makes it easy for others to understand.
+
+
+### 2. No Modularisation
+Functions/Methods should perform one task and one task only this helps them to kept short making it easier to maintian. This means that disclosure of needless information is avodied and emphasises the __*DRY (Don't Repeat Yourself)*__ belief eridicating repetitive code.
+
+### 3.Code Typos
+The slighest misspelling of a variable or a funcion can destroy your code and cause multiple run-time errors. They are also not easy to spot. Make sure variable names tahta re easy to spell and incorporate a good __*IDE (Integrated Devlopment Environment)*__ that can reduce spelling errors.
+
+### 4. Not Following Styling Guides
+Code with consitent style is much easier to read and understand, allowing the individual reading the code to understand blocks of code better. Ensure that your code is properly formatted and properly indented.
+
+### 5. Bad Estimates
+ it's important to avoid the temptation to imagine that you'll catch up with your schedule later without adding people to the project. If you fall behind schedule, it's because your time estimates were wrong. That means you need to make a new estimate of the length of the project, not blindly stick to an estimate that has already been proven wrong.
+
+ ### 6. Testing
+Testing is critical to assessing how well your code works. Software that has not been tested can lead to devestating results upon release. While testing often relies on automated tools and specialized staff, developers should take an active role in the process. They should test each iteration to ensure that it meets the necessary requirements,. Many developers would think testing as a waste of time and would lead to extra time bein added onto the timeframe given to complete the project but as they say _"Better safe then sorry"_.
```

Comment:
```suggestion
Testing is critical to assessing how well your code works. Software that has not been tested can lead to devastating results upon release. While testing often relies on automated tools and specialized staff, developers should take an active role in the process. They should test each iteration to ensure that it meets the necessary requirements. Many developers would think testing is a waste of time and would lead to extra time being added onto the time frame given to complete the project but as they say _"Better safe then sorry"_.
```